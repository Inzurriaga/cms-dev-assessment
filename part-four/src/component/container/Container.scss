  .container-nav {
    width: 100%;
    height: 100%;
    overflow: hidden;
  }


.container {
  width: 100%;
  height: 100%;
  transition: 1s;
  display: grid;
  box-sizing: border-box;
  padding: 50px;
  padding-bottom: 100px;
  grid-template-rows: minmax(0, 2fr) minmax(0, 1fr) minmax(0, 1fr) minmax(0, 2fr);
  grid-template-columns: minmax(0, 2fr) minmax(0, 1fr) minmax(0, 1fr) minmax(0, 2fr);
  grid-column-gap: 20px;
  grid-row-gap: 20px;
}

.container-exit {
  transform: translate( 0, 10%)
}

.container-enter {
  transform: translate( 0, 0 )
}

nav {
  position: absolute;
  bottom: 50px;
  right: 50px;

  button {
    background: none;
    cursor: pointer;
    font-family: 'Lato', sans-serif;
    font-weight: 400;
    outline: none;
    border: none;
    font-size: 1em;

    .prev-arrow {
      height: 10px;
      width: auto;
      margin: 0 10px;
    }
    
    .next-arrow {
      height: 10px;
      width: auto;
      transform: rotate(180deg);
      margin: 0 10px;
    }
  }
}