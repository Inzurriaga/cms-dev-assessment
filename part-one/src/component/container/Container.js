import React, { Component } from 'react';
import { connect } from 'react-redux';
import Card from '../card/Card';

export class Container extends Component {
  
  render() {
    const listingsCard = this.props.listings.data.map(list => {
      return (
        <Card key={ list.recid } data={ list }/>
      )
    });
    return (
      <section className='container'>
        {
          listingsCard
        }
      </section>
    );
  }
}

export const mapStateToProps = (state) => ({
  listings: state.listings
});

export default connect(mapStateToProps)(Container);