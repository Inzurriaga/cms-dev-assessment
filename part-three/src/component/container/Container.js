import React, { Component } from 'react';
import Card from '../card/Card';
import { connect } from 'react-redux';
import { ReactComponent as Arrow } from '../../arrow.svg';

export class Container extends Component {

  constructor() {
    super();
    this.state = {
      currentPage: 0,
      animation: ''
    };
  }

  changePage = (num) => {
    this.setState({
      animation: 'exit'
    }, () => this.nextPage(num));
  }

  nextPage = (num) => {
    setTimeout( () => {
    const nextPage = this.state.currentPage + num;
    if( nextPage === 4 ) {
      this.setState({
        currentPage: 0,
        animation: 'enter'
      });
    }else if( nextPage === -1 ) {
      this.setState({
        currentPage: 3,
        animation: 'enter'
      });
    }else {
      this.setState({
        currentPage: nextPage,
        animation: 'enter'
      });
    }
  }, 500);
  }

  render() {
    const { currentPage, animation } = this.state;
    const pageCards = this.props.listings[currentPage].map(( card, index )=> {
      return(
        <Card key={ card.recid } 
              gridPlacement={ `page-${ this.state.currentPage }-grid-${ index }` } 
              animation={ animation } 
              data={ card }/>
      )
    });
    return (
      <section className='container-nav'>
        <section className={ `container container-${animation}` }>
        {
          pageCards
        }
        </section>
        <nav>
          <button onClick={ () => this.changePage(-1) }>
            <Arrow className='prev-arrow'/>
            Prev.
          </button>
          <button onClick={ () => this.changePage(1) }>
            Next
            <Arrow className='next-arrow'/>
          </button>
        </nav>
      </section>
    );
  }
}

export const mapStateToProps = ( state ) => ({
  listings: state.listings
});

export default connect(mapStateToProps)(Container);