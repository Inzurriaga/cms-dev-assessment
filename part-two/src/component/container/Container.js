import React, { Component } from 'react';
import { connect } from 'react-redux';
import Card from '../card/Card';

export class Container extends Component {

  filterArray = () => {
    const { category, listings, offers, events } = this.props;
    let filterArray = [];
    switch(category) {
      case 'listings': 
        return filterArray.concat(listings.data);
      case 'events':
        return filterArray.concat(events.data);
      case 'offers':
        return filterArray.concat(offers.data);
      default:
        return filterArray.concat(listings.data, offers.data, events.data);
    }
  }

  render() {

    const updatedArray = this.filterArray();
    const listingsCard = updatedArray.map(list => {
      return (
        <Card key={list.recid} data={list} />
      );
    })

    return (
      <section className='container'>
        {
          listingsCard
        }
      </section>
    );
  }
}

export const mapStateToProps = (state) => ({
  listings: state.listings,
  offers: state.offers,
  events: state.events,
  category: state.category
});

export default connect(mapStateToProps)(Container);