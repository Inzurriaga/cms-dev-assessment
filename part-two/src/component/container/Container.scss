.container {
  padding: 10px;
  width: 100%;
  height: 100%;
  max-width: 1200px;
  display: grid;
  box-sizing: border-box;
  grid-template-rows: auto;
  grid-template-columns: repeat(4, 1fr);
  grid-row-gap: 25px;
  grid-column-gap: 25px;
}

@media screen and (max-width: 1025px) {

  .container {
    grid-template-columns: repeat(3, 1fr);
  }
}

@media screen and (max-width: 641px) {
  
  .container {
    grid-template-columns: 1fr;
  }
}
