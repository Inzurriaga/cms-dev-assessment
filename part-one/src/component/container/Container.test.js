import React from 'react';
import { shallow } from 'enzyme';
import { Container, mapStateToProps }from './Container';

const mockListings = { data: [] }

describe('Container', () => {

  describe('Container', () => {

    let wrapper;
    beforeEach(() => {
      wrapper = shallow(
        <Container listings={ mockListings } />
      );
    });

    it('should have a snap shot', () => {
      expect(wrapper).toMatchSnapshot()
    });
  });

  describe('mapStateToProps', () => {
    
    it('should return only what is needed', () => {
      const mockState = {
        listings: [],
        hello: 'hello'
      };
      const expected = {
        listings: []
      };
      const results = mapStateToProps(mockState);
      expect(results).toEqual(expected);
    });
  });
});