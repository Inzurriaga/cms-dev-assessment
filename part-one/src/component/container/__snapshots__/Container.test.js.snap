// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Container should have a snap shot 1`] = `ShallowWrapper {}`;
